src/profile-logic/function-info.js

```diff
@@ -42,35 +42,52 @@ export function stripFunctionArguments(functionCall: string): string {
 }
 
 export function removeTemplateInformation(functionName: string) {
+  // The result for this function. This works as an accumulator.
   let result = '';
+  // This keeps the depth in the tree of anchorness.
   let depth = 0;
-  let start = 0; // Start of a segment we'd like to keep
+  // This keeps the depth start of a template we'd like to remove. This is
+  // useful to find the end of this template. This is null if we're not removing
+  // any template yet.
+  let templateStartDepth: null | number = null;
```

Comment:
I realized the algorithm was increasing depth only in some conditions, but decreasing depth always. So I changed the algorithm to support this case even though we didn't see that in the past. This is causing an issue if we skip a "false positive template information" inside a real template information. I added a test for this case.